nemo/collections/nlp/modules/common/token_classifier.py

```diff
@@ -50,7 +50,89 @@ def __init__(
         self,
         hidden_size: int,
         num_classes: int,
-        num_layers: int = 1,
+        num_layers: int = 2,
+        activation: str = 'relu',
+        log_softmax: bool = True,
+        dropout: float = 0.0,
+        use_transformer_init: bool = True,
+    ) -> None:
+
+        """
+        Initializes the Token Classifier module.
+
+        Args:
+            hidden_size: the size of the hidden dimension
+            num_classes: number of classes
+            num_layers: number of fully connected layers in the multilayer perceptron (MLP)
+            activation: activation to usee between fully connected layers in the MLP
+            log_softmax: whether to apply softmax to the output of the MLP
+            dropout: dropout to apply to the input hidden states
+            use_transformer_init: whether to initialize the weights of the classifier head with the same approach used in Transformer
+        """
+        super().__init__()
+        self.mlp = MultiLayerPerceptron(
+            hidden_size, num_classes, num_layers=num_layers, activation=activation, log_softmax=log_softmax
+        )
+        self.dropout = nn.Dropout(dropout)
+        if use_transformer_init:
+            self.apply(lambda module: transformer_weights_init(module, xavier=False))
+
+    @typecheck()
+    def forward(self, hidden_states):
+        """
+        Performs the forward step of the module.
+        Args:
+            hidden_states: batch of hidden states (for example, from the BERT encoder module)
+                [BATCH_SIZE x SEQ_LENGTH x HIDDEN_SIZE]
+        Returns: logits value for each class [BATCH_SIZE x SEQ_LENGTH x NUM_CLASSES]
+        """
+        hidden_states = self.dropout(hidden_states)
+        logits = self.mlp(hidden_states)
+        return logits
+
+    def save_to(self, save_path: str):
+        """
+        Saves the module to the specified path.
+        Args:
+            save_path: Path to where to save the module.
+        """
+        pass
+
+    @classmethod
+    def restore_from(cls, restore_path: str):
+        """
+        Restores the module from the specified path.
+        Args:
+            restore_path: Path to restore the module from.
+        """
+        pass
+
+
+@experimental
+class BertPretrainingTokenClassifier(NeuralModule):
+    """
+    A module to perform token level classification tasks for Bert pretraining.
+    """
+
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        """
+        Returns definitions of module input ports.
+        """
+        return {"hidden_states": NeuralType(('B', 'T', 'D'), ChannelType())}
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        """
+        Returns definitions of module output ports.
+        """
+        return {"logits": NeuralType(('B', 'T', 'C'), LogitsType())}
+
+    def __init__(
+        self,
+        hidden_size: int,
+        num_classes: int,
+        num_layers: int = 2,
```

Comment:
num_layers was 1 for BERTtokenclassifier

Reply:
changed